File: tests/testdata/codegen/deferred-outputs-pp/dotnet/Second.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Random = Pulumi.Random;

namespace Components
{
    public class SecondArgs : global::Pulumi.ResourceArgs
    {
        [Input("petName")]
        public Input<string> PetName { get; set; } = null!;
    }

    public class Second : global::Pulumi.ComponentResource
    {
        [Output("passwordLength")]
        public Output<int> PasswordLength { get; private set; }
        public Second(string name, SecondArgs args, ComponentResourceOptions? opts = null)
            : base("components:index:Second", name, args, opts)
        {
            var randomPet = new Random.RandomPet($"{name}-randomPet", new()
            {
                Length = args.PetName.Length,
            }, new CustomResourceOptions
            {
                Parent = this,
            });

            var password = new Random.RandomPassword($"{name}-password", new()
            {
                Length = 16,
                Special = true,
                Numeric = false,
            }, new CustomResourceOptions
            {
                Parent = this,
            });

            this.PasswordLength = password.Length;

            this.RegisterOutputs(new Dictionary<string, object?>
            {
                ["passwordLength"] = password.Length,
            });
        }
    }
}

File: tests/testdata/codegen/deferred-outputs-pp/python/second.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_random as random

class SecondArgs(TypedDict, total=False):
    petName: Input[str]

class Second(pulumi.ComponentResource):
    def __init__(self, name: str, args: SecondArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:Second", name, args, opts)

        random_pet = random.RandomPet(f"{name}-randomPet", length=len(args["petName"]),
        opts = pulumi.ResourceOptions(parent=self))

        password = random.RandomPassword(f"{name}-password",
            length=16,
            special=True,
            numeric=False,
            opts = pulumi.ResourceOptions(parent=self))

        self.passwordLength = password.length
        self.register_outputs({
            'passwordLength': password.length
        })